datastore/mongo/event.go

```diff
@@ -37,7 +37,7 @@ func (db *eventRepo) CreateEvent(ctx context.Context, message *datastore.Event)
 	message.ID = primitive.NewObjectID()
 
 	if util.IsStringEmpty(message.ProviderID) {
-		message.ProviderID = message.EndpointID
+		message.ProviderID = uuid.NewString()
```

Comment:
why are we setting this to a random uuid4

Reply:
What is this field used for though? I don't think I've ever used it Cc @Subomi

Reply:
Yeah, we should deprecate that field.